doc/metrics.md

```diff
@@ -0,0 +1,286 @@
+
+#### Filtering by URL and method
+
+To get URL and method metrics and metrics data, use the
+`getClientMetricsByUrlAndMethod()` and `getClientMetricsDataByUrlAndMethod()`
+methods in the `Metric` class.
+
+Both of these take as arguments the `MetricRegistry`, a string url, and a
+string HTTP method. If no url or method is provided, will return all
+url-and-method metrics. If no metrics are registered for that url and method,
+returns an empty list.
+
+`getClientMetricsByUrlAndMethod()` returns a list of `UrlAndMethodClientTimer`
+objects. `getClientMetricsDataByUrlAndMethod` returns a list of
+`UrlAndMethodClientMetricData` objects.
+
+#### Filtering by metric-id
+
+To get metric-id metrics and metrics data, use the
+`getClientMetricsByMetricId()` and `getClientMetricsDataByMetricId()` methods
+in the `Metric` class.
+
+Both of these take as arguments the `MetricRegistry` and a metric-id - as an
+array of strings. If no metric-id is provided, will return all metric-id
+metrics. If no metrics are registered for that metric-id, returns an empty
+list.
+
+`getClientMetricsByMetricId()` returns a list of `MetricIdClientTimer`
+objects. `getClientMetricsDataByMetricId()` returns a list of
+`MetricIdClientMetricData` objects.
+
+## Clojure API
+
+### Creating a client with metrics
+To use metrics with the Clojure client, pass a `MetricRegistry` to it as part
+of the options map:
+
+```clojure
+(async/create-client {:metric-registry (MetricRegistry.)})
+```
+
+(the same works with the `sync` client).
+
+Any client that is created with a `MetricRegistry` will automatically have
+`url` and `url-and-method` metrics registered.
+
+`get-client-metric-registry` protocol function can be called on a client to
+get the `MetricRegistry` from it.
+
+### Setting a metric-id
+
+In addition to the `url` and `url-and-method` metrics, it is possible to set a
+`metric-id` for a request that will create additional metrics.
+
+For the Clojure API, a `metric-id` is a vector of keywords or strings. Either
+is supported by the API, however, if special characters are needed, use
+strings.  Note than even when specified as a vector of keywords in the request
+option, the metric-id will be returned as a vector of strings in the metrics
+data.
+
+To set a `metric-id` for a request, include it as an option in the request
+options map.
+
+```clojure
+(common/get client "https://foobar.com" {:metric-id [:foo :bar :baz]})
+(common/get client "https://foobar.com" {:metric-id ["f/o/o"]})
+```
+
+### Filtering metrics
+
+To get all `Timer` objects registered for a `MetricRegistry`, use the
+`get-client-metrics` function in the `metrics` namespace.  This takes the
+`MetricRegistry` as an argument and returns a map with three keys: `:url`
+`:url-and-method`, and `:metric-id`. Under each of these keys is a sequence of
+`Timer` objects of the corresponding type. If there are no timers of a certain
+type, the sequence will be empty.
```

Comment:
I think it might be useful to throw in an example of what the result object looks like, and maybe also mention the namespace/var of the schema that it conforms to?